### ISSUE TITLE
Release Pipeline Continues to not publish galaxy release

### ISSUE THREAD

**Issue by thedoubl3j** [ansible/awx#14681]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

When releasing awx, we are continuing to see issues with the galaxy piece of the promote workflow. 

This issue is no longer key related but either located in the curl command (my guess is the version that it is checking for) or in the workflow around what version of the collection it is building in the pipeline. 

The most recent run for 23.5.0 did not publish the collection and said in its out from the curl check that the collection already existed (it did not, this version was manually pushed again)

### AWX version

devel

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Installation method

N/A

### Modifications

no

### Ansible version

n/a

### Operating system

n/a

### Steps to reproduce

Run the stage and promote workflows with a new version of awx that does not already exist. 

### Expected results

release is published across all platforms (galaxy/quay etc)

### Actual results

Step to publish collection to galaxy is skipped due to the version already existing. 

### Additional information

line where the curl command is run in the workflow https://github.com/ansible/awx/blob/23.5.0/.github/workflows/promote.yml#L49

**Comment by Denney-tech** [comment]
The code appears to be looking for a 302 redirect as a confirmation of upload, but the url might be an old style that's getting a 302 regardless. Perhaps it would be better to check the api for the existence of the version which should imply the collection was successfully uploaded or otherwise check the correct download url.

E.g. https://galaxy.ansible.com/download/awx-awx-23.5.0.tar.gz may redirect to https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/awx-awx-23.5.0.tar.gz, and this redirect may be happening even if the artifact doesn't exist.

Instead, it could check the api to see if that version of the collection exists, which should imply that the artifact was successfully uploaded: https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/index/awx/awx/versions/23.5.0/ 

Edit: After some testing, it may be better to add the L flag to make sure redirects are followed and then get the final page's status code, expecting a 200 instead of a 302.

Example suggestions:

1. Follow redirects and expect 200.
``` diff
-           if [ "$(curl --head -sw '%{http_code}' https://galaxy.ansible.com/download/${{ env.collection_namespace }}-awx-${{ github.event.release.tag_name }}.tar.gz | tail -1)" == "302" ] ; then \
+           if [ "$(curl --head -sLw '%{http_code}' https://galaxy.ansible.com/download/${{ env.collection_namespace }}-awx-${{ github.event.release.tag_name }}.tar.gz | tail -1)" == "200" ] ; then \
```
2. Follow redirects and expect 200, use api published download_url.
``` diff
-           if [ "$(curl --head -sw '%{http_code}' https://galaxy.ansible.com/download/${{ env.collection_namespace }}-awx-${{ github.event.release.tag_name }}.tar.gz | tail -1)" == "302" ] ; then \
+           if [ "$(curl --head -sLw '%{http_code}' https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/artifacts/${{ env.collection_namespace }}-awx-${{ github.event.release.tag_name }}.tar.gz | tail -1)" == "200" ] ; then \
```
3. Follow redirects and expect 200, version only check instead of download check.
``` diff
-           if [ "$(curl --head -sw '%{http_code}' https://galaxy.ansible.com/download/${{ env.collection_namespace }}-awx-${{ github.event.release.tag_name }}.tar.gz | tail -1)" == "302" ] ; then \
+           if [ "$(curl --head -sLw '%{http_code}' https://galaxy.ansible.com/api/v3/plugin/ansible/content/published/collections/index/${{ env.collection_namespace }}/awx/versions/${{ github.event.release.tag_name }}/ | tail -1)" == "200" ] ; then \
```

**Comment by thedoubl3j** [comment]
@Denney-tech thanks for digging in, looks like we had an outstanding PR that never merged that actually fixed this :expressionless:
